pvlib/iotools/pvgis.py

```diff
@@ -410,6 +410,8 @@ def _coerce_and_roll_tmy(tmy_data, tz, year):
         np.roll(tmy_data, tz, axis=0),
         columns=tmy_data.columns,
         index=new_index)
+    new_tmy_data = \  # GH 2399
+        new_tmy_data.astype(dtype=dict(zip(tmy_data.columns, tmy_data.dtypes)))
```

Comment:
```suggestion
    # GH 2399
    new_tmy_data = \
        new_tmy_data.astype(dtype=dict(zip(tmy_data.columns, tmy_data.dtypes)))
```

Sorry my last suggestion broke this :(